src/main/kotlin/com/mtereshchuk/account/service/AccountServiceImpl.kt

```diff
@@ -0,0 +1,199 @@
+package com.mtereshchuk.account.service
+
+import com.mtereshchuk.account.model.*
+import com.mtereshchuk.account.util.Failure
+import com.mtereshchuk.account.util.Result
+import com.mtereshchuk.account.util.Success
+import com.mtereshchuk.account.util.notIn
+import java.math.BigDecimal
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.locks.Lock
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * @author mtereshchuk
+ */
+data class LockAndAccount(
+    val lock: Lock,
+    val account: Account
+)
+
+class AccountServiceImpl : AccountService {
+    private val idGenerator = AtomicLong(1)
+    private val accounts = ConcurrentHashMap<AccountId, LockAndAccount>()
```

Comment:
Abstractions such as `AccountDao` or `AccountRepository` is redundant due to the in-memory storage